src/facade/dragonfly_connection.cc

```diff
@@ -724,11 +725,29 @@ auto Connection::ParseMemcache() -> ParserStatus {
       break;
     }
 
+    auto parse_value = [&consumed, &cmd](std::string_view str) -> std::optional<std::string_view> {
+      std::string_view value = str.substr(consumed, cmd.bytes_len);
+      const absl::flat_hash_set<char> any_of{' ', '\r', '\n'};
+      if (any_of.contains(value.back())) {
+        return {};
+      }
+      auto sub_value = str.substr(consumed);
```

Comment:
what's the illegal state that you are checking here?

Reply:
Basically, the problem is that we receive a string of:

`set mykey 0 0 3\r\nkostas\r\n`. At the point we parse the payload/value the start of the marker points to `k`. We need to check if the substring from the marker till the logical end (cmd.bytes_len). If this ends, with any of the `\r\n` it means we undershoot (we passed a string that is less than the specified length. Example is `set myset 0 0 4 foo -- we told we are passing a value of 4 characters but we passed three). The second check is for the opposite, if we passed a string that is greater than the specified length (example here is set myset 0 0 3 foobar -- value has 6 chars instead of 3). 

Now I am thinking about it, when we reach to the parsing point, the substring will always end with `\r\n` so it suffices to only compare the lengths. I will push a quick fix that simplifies the above chore.